### PR TITLE
add gainsight-px doc

### PR DESCRIPTION
**Pull request by arajguha**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: destinations/marketing/gainsight_px.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+| **Device mode**     | -             | -             | -             |
+| **Cloud mode**      | **Supported** | **Supported** | **Supported** |
+
+{% hint style="info" %}
+To know more about the difference between Cloud mode and Device mode in RudderStack, read the [**RudderStack connection modes**](https://docs.rudderstack.com/get-started/rudderstack-connection-modes) guide.
+{% endhint %}
+
+Once you've confirmed that the source platform supports sending events to RudderStack, follow these steps:
+
+- From your [**RudderStack dashboard**](https://app.rudderlabs.com/), add the source and select **Gainsight PX** from the list of supported destinations.
```
Comment (ameypv-rudder) [inline destinations/marketing/gainsight_px.md:36]: Please confirm the destination name here. I've changed it from Gainsight to Gainsight PX (the previous Gainsight CS destination also had Gainsight as the destination name so we might have to revisit that too, if required.) cc @arnab-p @arajguha

File: destinations/marketing/gainsight_px.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+{% hint style="info" %}
+To add a source, follow our guide on [**How to Add a Source and Destination in RudderStack**](https://docs.rudderstack.com/how-to-guides/adding-source-and-destination-rudderstack).
+{% endhint %}
+
+- Assign a name to this destination and click on **Next**. You should then see the following **Connection Settings** screen:
+
+![Connection screen for Gainsight PX](../../.gitbook/assets/gainsight-px-connection-settings-1.png)
```
Comment (ameypv-rudder) [inline destinations/marketing/gainsight_px.md:44]: Should we show the full connection settings screen including the mapping sections too?

File: destinations/marketing/gainsight_px.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+### Custom Attributes
+
+You can send Custom Attributes for both the `User` and `Account` objects using RudderStack.
+
+To create custom attributes in Gainsight PX, go to your Gainsight dashboard and navigate to **Administration** - **Attributes**. Then, click on the **New Attribute** button as shown in the image below:
+
+![Create Custom Attribute](../../.gitbook/assets/gainsight-px-create-attribute.png)
+
+You can then provide this mapping in the RudderStack dashboard.
```
Comment (ameypv-rudder) [inline destinations/marketing/gainsight_px.md:67]: Including the mappings section in the screenshot will add clarity to this sentence. We can show the mapping bit here too, for better clarity.

File: destinations/marketing/gainsight_px.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+## Identify
+
+RudderStack creates a **`User`** object in Gainsight PX for the `identify` call. Gainsight uses the `identifyId` field as the unique identifier for this object. RudderStack will then map the `userId` from the message to this `identifyId`. 
+
+{% hint style="info" %}
+If `userId` is not present, `anonymousId` will be used instead.
+{% endhint %}
+
+RudderStack supports all the default attributes for the **`User`** object. Additionally, for the custom attributes you can map the RudderStack payload key to the Gainsight PX key in the RudderStack dashboard as shown:
```
Comment (ameypv-rudder) [inline destinations/marketing/gainsight_px.md:78]: Can we rephrase this sentence to:

RudderStack supports all the default attributes for the **`User`** object. Additionally, for the custom attributes you can map the RudderStack payload attributes to the Gainsight PX attributes in the RudderStack dashboard under the **User Attribute Mapping** section, as shown:

cc @arajguha

File: destinations/marketing/gainsight_px.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+![User Custom Attribute Mapping](../../.gitbook/assets/gainsight-px-user-attribute-mapping-1.png)
+
+
+{% hint style="warning" %}
+If the mapping is not provided, all the attributes apart from the default ones are dropped.
```
Comment (ameypv-rudder) [inline destinations/marketing/gainsight_px.md:84]: Can we rephrase this to:

"RudderStack will drop the custom attributes which are not defined in the mapping and send the rest of the attributes to Gainsight PX."

File: destinations/marketing/gainsight_px.md (hunk 4 above)
Comment (ameypv-rudder) [reply destinations/marketing/gainsight_px.md:78]: Also, these Gainsight PX attributes have to be defined in Gainsight PX as described in the **Custom Attributes** section, right?

File: destinations/marketing/gainsight_px.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+rudderanalytics.identify("sample-user-id", {
+  email: "user@domain.com",
+  name: "Some User",
+  gender: "M",
+  countryName: "INDIA",
+  countryCode: "IN",
+  city: "CHENNAI",
+  score: 100,
+  title: "Doctor",
+});
```
Comment (ameypv-rudder) [inline destinations/marketing/gainsight_px.md:99]: Can we include a custom field in the snippet here? In the screenshot, `hobbyCustomField` is mapped to the `hobby` custom attribute. 

We can also include a description similar to the one we have in the Group call section:

In the above example, `hobbyCustomField` is a custom field. You will need to provide the mapping for `hobbyCustomField` to the corresponding Custom Attribute name (`hobby`) in Gainsight PX. If this mapping is not provided, RudderStack will drop the `hobbyCustomField` attribute and send the other attributes to Gainsight.

File: destinations/marketing/gainsight_px.md (hunk 1 above)
Comment (arajguha) [reply destinations/marketing/gainsight_px.md:36]: @ameypv-rudder  Yes. The name of this destination would be Gainsight PX and the previous one was Gainsight CS. They are supposed to be different destinations.

File: destinations/marketing/gainsight_px.md (hunk 4 above)
Comment (arajguha) [reply destinations/marketing/gainsight_px.md:78]: @ameypv-rudder  Yes. The Custom Attributes need to be created in Gainsight PX first and then the mapping must be given here.

File: destinations/marketing/gainsight_px.md (hunk 5 above)
Comment (arajguha) [reply destinations/marketing/gainsight_px.md:84]: Sure, this sounds better.

File: destinations/marketing/gainsight_px.md (hunk 1 above)
Comment (ameypv-rudder) [reply destinations/marketing/gainsight_px.md:36]: Great, will make the change in the CS destination too.

**Review by ameypv-rudder — APPROVED** [review]
Made the necessary changes as per our discussion, @arajguha :)  Good to go!

**Review by arnab-p — APPROVED** [review]
